client/src/Controller/Admin/StatsController.php

```diff
@@ -288,24 +280,41 @@ public function downloadAssetsTotalValues(): Response
     }
 
     /**
-     * @Route("/reports/downloadOldAdminUsersCsv", name="admin_old_user_account_report")
+     * @Route("/reports/inactive-admin-users-report", name="inactive_admin_users_report")
      * @Security("is_granted('ROLE_SUPER_ADMIN')")
-     *
+     * @Template("@App/Admin/Stats/inactiveAdminUsersReport.html.twig")
      */
-    public function downloadOldAdminUsersCsv(): Response
+    public function inactiveAdminUsersReport(Request $request): array|Response
     {
-        $fileName = 'inactiveAdminUsers.csv';
-        $reportData = $this->statsApi->getOldAdminUsers();
-        $csv = $this->inactiveAdminUserCsvGenerator->generateOldAdminUsersCsv($reportData);
+        $form = $this->createForm(InactiveAdminReportFilterType::class);
 
-        return $this->csvResponseGeneration($fileName, $csv);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            try {
+                $inactivityPeriod = $form->get('inactivityPeriod')->getData();
+
+                $append = '?inactivityPeriod='.$inactivityPeriod;
+
+                $fileName = 'inactiveAdminUsers.csv';
+                $reportData = $this->statsApi->getInactiveAdminUsers($append);
+                $csv = $this->inactiveAdminUserCsvGenerator->generateOldAdminUsersCsv($reportData);
```

Comment:
All other reference to OldAdmin have been removed, should probably change this one also.